service/matching/matching_engine_test.go

```diff
@@ -3397,14 +3395,18 @@ func (m *testTaskManager) CreateTasks(
 	taskType := request.TaskQueueInfo.Data.TaskType
 	rangeID := request.TaskQueueInfo.RangeID
 
-	// Randomly returns an error
+	// Randomly returns a ConditionFailedError
 	if m.generateErrorRandomly() {
 		return nil, &persistence.ConditionFailedError{
 			Msg: fmt.Sprintf("Failed to create task. TaskQueue: %v, taskQueueType: %v, rangeID: %v, db rangeID: %v",
 				taskQueue, taskType, rangeID, rangeID),
 		}
 	}
 
+	if m.dbServiceError {
```

Comment:
I guess I meant this should also be a random error using the same mechanism, not every time